apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1227,11 +1227,34 @@ async function validateAssessment(
               'Cannot specify "maxPoints" for a question if "autoPoints", "manualPoints" or "maxAutoPoints" are specified',
             );
           }
+
           if (Array.isArray(alternative.autoPoints ?? alternative.points)) {
             errors.push(
               'Cannot specify "points" or "autoPoints" as a list for a question in a "Homework" assessment',
             );
           }
+
+          if (!courseInstanceExpired) {
+            if (
+              alternative.points !== undefined &&
+              alternative.points === 0 &&
+              alternative.maxPoints !== undefined &&
+              alternative.maxPoints > 0
+            ) {
+              warnings.push('Specifying "points": 0 when "maxPoints" > 0 is probably a mistake');
+            }
+
+            if (
+              alternative.autoPoints !== undefined &&
+              alternative.autoPoints === 0 &&
+              alternative.maxAutoPoints !== undefined &&
+              alternative.maxAutoPoints > 0
+            ) {
+              warnings.push(
+                'Specifying "autoPoints": 0 when "maxAutoPoints" > 0 is probably a mistake',
+              );
+            }
```

Comment:
I erred on the side of caution by making these warnings. Can anyone think of _any_ valid reason one would want this configuration? If not, it's probably safe to make this an error.

Reply:
I'm on board with this being an error. Can we trigger this warning/error only for assessments in `accessibleInFuture` course instances? (Apologies if that's what we are already doing)

Reply:
I agree, this should be an error.

Reply:
The `if (!courseInstanceExpired) ...` conditional is essentially the same as checking `accessibleInFuture`.

I'll make this be an error instead.

Reply:
Changed to an error in 30b1f1c.

Reply:
Thanks for clarifying about `courseInstanceExpired`